File: backend/app/schemas/support.py
"""
Schemas — signalements de bug.
"""

from datetime import datetime
from pydantic import BaseModel, Field


class BugReportCreate(BaseModel):
    subject: str = Field(min_length=1, max_length=100)
    body: str = Field(min_length=1, max_length=2000)
    page_context: str | None = Field(default=None, max_length=200)


class BugReportOut(BaseModel):
    id: int
    username: str
    subject: str
    body: str
    page_context: str | None = None
    created_at: datetime
    resolved_at: datetime | None = None

File: web/src/api/support.ts
import api from "./client";

export interface BugReportCreate {
    subject: string;
    body: string;
    page_context?: string;
}

export interface BugReportOut {
    id: number;
    username: string;
    subject: string;
    body: string;
    page_context: string | null;
    created_at: string;
    resolved_at: string | null;
}

export const supportApi = {
    submitBugReport: (data: BugReportCreate) =>
        api.post<BugReportOut>("/support/bug-reports", data).then((r) => r.data),
};
